specs/core/0_beacon-chain.md

```diff
@@ -1382,16 +1378,7 @@ Note: All functions in this section mutate `state`.
 def activate_validator(state: BeaconState, index: int, genesis: bool) -> None:
     validator = state.validator_registry[index]
 
-    validator.activation_slot = GENESIS_SLOT if genesis else (state.slot + ENTRY_EXIT_DELAY)
-    state.validator_registry_delta_chain_tip = hash_tree_root(
-        ValidatorRegistryDeltaBlock(
-            latest_registry_delta_root=state.validator_registry_delta_chain_tip,
-            validator_index=index,
-            pubkey=validator.pubkey,
-            slot=validator.activation_slot,
-            flag=ACTIVATION,
-        )
-    )
+    validator.activation_slot = GENESIS_SLOT if genesis else (state.slot - state.slot % EPOCH_LENGTH + ENTRY_EXIT_DELAY)
```

Comment:
I'm tempted to add a helper -- `epoch_start_slot(slot)` -- that takes input `slot` and returns `slot - slot % EPOCH_LENGTH`.  We're doing this in many places at this point and should increase readability.

Reply:
I just added a helper `entry_exit_effect_slot` which I think even more closely targets the computation that's being repeated a bunch of times.